src/data/real/dedekind_cut.lean

```diff
@@ -0,0 +1,403 @@
+/-
+Copyright (c) 2022 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import algebra.order.archimedean
+import data.set.pointwise
+import order.upper_lower
+
+/-!
+# Dedeking cuts
+
+A basic theory of Dedekind cuts, used in a construction of the reals.
```

Comment:
I think it might be worth noting that reals are defined elsewhere in terms of Cauchy sequences.